client/src/controllers/BreadcrumbsController.ts

```diff
@@ -0,0 +1,132 @@
+import { Controller } from '@hotwired/stimulus';
+
+export class BreadcrumbsController extends Controller {
+  static targets = ['w-breadcrumbs', 'toggle'];
+  private toggleTarget!: HTMLElement;
+  private breadcrumbsTargets!: Element[];
+  
+  private open = false;
+  private mouseExitedToggle = true;
+  private keepOpen = false;
+
+  private cssClass = {
```

Comment:
Recommend we use the built in system of classes with Stimulus. 

https://stimulus.hotwired.dev/reference/css-classes

Note .these do not support default values, but that's intentional..this means the CSS classes needed are forced to be declared on the html element. For colocation of behaviour and CSS stuff being all on the html and not declared in JS.